docs/architecture/Utils.mdx

```diff
@@ -0,0 +1,28 @@
+---
```

Comment:
we defs need to do better with these specific docs... a big todo to expand on them!

Reply:
@bradzacher anything in particular you're thinking of? And/or, up for filing an issue?

Reply:
Right now it just provides a very high-level overview of the top-level exported names in the package.

Really we would want to use something like https://api-extractor.com/ to do some form of generation so that people have an in-depth reference of what's included.

Also some examples of how to use some things or why to use some things.

For example - people might not know why you'd use the `TSESLint` export from the package (it's because `@types/eslint` is built on top of the base `@types/estree` types which don't include our AST nodes)

Reply:
👍 filed #6119